nemo/collections/asr/data_layer.py

```diff
@@ -425,3 +428,165 @@ def dataset(self):
+            transcript or just the transcript.
+            Defaults to True.
+        drop_last (bool): See PyTorch DataLoader.
+            Defaults to False.
+        shuffle (bool): See PyTorch DataLoader.
+            Defaults to True.
+        num_workers (int): See PyTorch DataLoader.
+            Defaults to 0.
+        augmentor (dict): Optional dictionary of str -> kwargs (dict)
+            which is parsed and used to initialize an AudioAugmentor.
+            Note: It is crucial that each individual augmentation has
+            a keyword `prob`, that defines a float probability in the
+            the range [0, 1] of this augmentation being applied.
+            If this keyword is not present, then the augmentation is
+            disabled and a warning is logged.
+    """
+
+    @property
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        """
+        return {
+            # 'audio_signal': NeuralType({0: AxisType(BatchTag), 1: AxisType(TimeTag)}),
+            # 'a_sig_length': NeuralType({0: AxisType(BatchTag)}),
+            # 'label': NeuralType({0: AxisType(BatchTag)}),
+            # 'label_length': NeuralType({0: AxisType(BatchTag)}),
+            'audio_signal': NeuralType(('B', 'T'), AudioSignal(freq=self._sample_rate)),
+            'a_sig_length': NeuralType(tuple('B'), LengthsType()),
+            'label': NeuralType(tuple('B'), LabelsType()),
+            'label_length': NeuralType(tuple('B'), LengthsType()),
+        }
+
+    def __init__(
+        self,
+        *,
+        manifest_filepath: str,
+        labels: List[str],
+        batch_size: int,
+        sample_rate: int = 16000,
+        int_values: bool = False,
+        num_workers: int = 0,
+        shuffle: bool = True,
+        min_duration: Optional[float] = 0.1,
+        max_duration: Optional[float] = None,
+        trim_silence: bool = False,
+        drop_last: bool = False,
+        load_audio: bool = True,
+        augmentor: Optional[Dict[str, Dict[str, Any]]] = None,
+    ):
+        super(AudioToSpeechLabelDataLayer, self).__init__()
+
+        self._manifest_filepath = manifest_filepath
+        self._labels = labels
+        self._sample_rate = sample_rate
+
+        if augmentor is not None:
+            augmentor = self._process_augmentations(augmentor)
+
+        self._featurizer = WaveformFeaturizer(sample_rate=sample_rate, int_values=int_values, augmentor=augmentor)
+
+        dataset_params = {
+            'manifest_filepath': manifest_filepath,
+            'labels': labels,
+            'featurizer': self._featurizer,
+            'max_duration': max_duration,
+            'min_duration': min_duration,
+            'trim': trim_silence,
+            'load_audio': load_audio,
+        }
+        self._dataset = AudioLabelDataset(**dataset_params)
+
+        # Set up data loader
+        if self._placement == DeviceType.AllGpu:
+            nemo.logging.info("Parallelizing Datalayer.")
```

Comment:
Please switch all instances of nemo.logging to just logging and adding either `logging = nemo.logging` or `from nemo import logging`. Can you also change all 12 instances? But there is no need to change what was already there. I will change it in my PR.

Reply:
Will change nemo.logging to logging. However, 12 instances of nemo.logging? Not taking into account the earlier data layers, I can see only 2 logging commands in AudioToSpeechDataLayer. Does the 12 refer to changes on all of the files?

Another thing is do I keep both logging and nemo.logging for files that were already there?

Reply:
You don't have to change any logging statements that are already present. But for the new ones that you added, can you change them?
There should be 12 across all changed files.